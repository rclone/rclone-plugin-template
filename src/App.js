import React from 'react';
import './App.css';
import rcloneLogo from "./assets/rclone/logo_symbol.png"

function App() {
  return (
    <div data-test="appComponent">
      <img src={rcloneLogo} alt={"rclone"}/>
      Welcome to your first rclone webui plugin.
    </div>
  );
}

export default App;
