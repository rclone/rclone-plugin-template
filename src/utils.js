import React from "react";

export const convertToHTML = (text) => {
    return (<pre>{text}</pre>);
}

export const addToArrayImmutable = (array, value) => {
    return [...array, {value}];
}

export const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
};