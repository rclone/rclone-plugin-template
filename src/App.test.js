import {shallow} from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";
import App from "./App";
import {findByTestAttr} from "./utils";


const setUp = ( props = {}) => {
  const component = shallow(<App {...props}/>);

  return component;
}

describe('App', function () {
  describe('renders', function () {
    let wrapper;
    beforeEach(()=>{
      const props = {};
      wrapper = setUp(props);
    })

    it('should render without crashing', function () {
      const component = findByTestAttr(wrapper, "appComponent");
      expect(component).toHaveLength(1);
    });

    it('should match snapshot', function () {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});