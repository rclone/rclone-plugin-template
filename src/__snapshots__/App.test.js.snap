// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders should match snapshot 1`] = `
<div
  data-test="appComponent"
>
  <img
    alt="rclone"
    src="logo_symbol.png"
  />
  Welcome to your first rclone webui plugin.
</div>
`;
